usecases/ingestion_usecase.go

```diff
@@ -45,9 +46,13 @@ func (usecase *IngestionUseCase) IngestObjects(ctx context.Context, organization
 	if err := usecase.enforceSecurity.CanIngest(organizationId); err != nil {
 		return err
 	}
-	return usecase.transactionFactory.TransactionInOrgSchema(ctx, organizationId, func(tx repositories.Executor) error {
-		return usecase.ingestionRepository.IngestObjects(ctx, tx, payloads, table, logger)
-	})
+
+	ingestClosure := func() error {
+		return usecase.transactionFactory.TransactionInOrgSchema(ctx, organizationId, func(tx repositories.Executor) error {
+			return usecase.ingestionRepository.IngestObjects(ctx, tx, payloads, table, logger)
+		})
+	}
+	return retryIngestion(ctx, ingestClosure)
```

Comment:
By curiosity, do we need to retry outside the transaction to make it "works" or is it independant ? (I must admit I fail to grasp why retry is needed here)

Reply:
The idea is the following:
Everything that's done inside the transaction, either is committed or everything is rolled back. It will be rolled back if there is an error (for instance an insertion conflict error).
It's fine and expected that there may be such insertion conflicts, and they may or may not be transient (they are if it's just a concurrency issue).
The thing to do then is to retry everything that's done within the transaction, from the beginning.